Updated content (#13084)
docker/docker.github.io: docker-hub/access-tokens.md

Docker Hub lets you create personal access tokens as alternatives to your password. You can use tokens to access Hub images from the Docker CLI.

Using personal access tokens provides some advantages over a password:
* You can investigate the last usage of the access token and disable or delete it if you find any suspicious activity.
* When logged in with an access token, you can't perform any admin activity on the account, including changing the password.

>**Important**
>
> Treat access tokens like your password and keep them secret. Store your
> tokens securely (for example, in a credential manager).
{: .important}

> Access tokens are valuable for building integrations, since you can issue

Reviewer: nit: since > as

Author: updated

multiple tokens &ndash; one for each integration &ndash; and revoke them at
any time.

   >**Note**
   >
   > If you have [two-factor authentication (2FA)](/docker-hub/2fa) enabled on
   > your account, you must create at least one personal access token. Otherwise,
   > you will be unable to log in to your account from the Docker CLI.

## Create an access token

You can create as many tokens as you need.

1. Log in to [hub.docker.com](https://hub.docker.com){: target="_blank" rel="noopener" class="_"}.

> 2. Click on your username in the top right corner and select **[Account Settings](https://hub-stage.docker.com/settings/general)**.

Reviewer: Update stage URL

Author: Updated

> 3. Select **[Security](https://hub-stage.docker.com/settings/security) > New Access Token**.

Reviewer: @jerae-duffin Could you change the stage links to Prod?

Author: Good catch, thank you!

4. Add a description for your token. Use something that indicates where
the token will be used, or set a purpose for the token.

      ![PAT Menu](images/hub-create-token.png){:width="700px"}

5. You can view the following permissions from the drop-down:

    | Features   | Free | Pro | Team |
    | --------------------- | ---- | ----- | ----- |
    | Personal Access Tokens (read, write, delete scope only)*    |  x   |   x   |   x   |
    | Scoped Personal Access Tokens**  |      | x   |   x   |
    | Unlimited Personal Access Tokens   |      |       |   x   |

>       >If you would like more information about a Pro or a Team plan, see [Docker Pricing](https://www.docker.com/pricing?utm_source=docker&utm_medium=webreferral&utm_campaign=docs_driven_upgrade){: target="_blank" rel="noopener" class="_"}.

Reviewer: could you change it to 'For more information...'

Author: Updated

> 6. Copy the token that appears on the screen. Do this now:

Reviewer: Remove: Do this now. 
Suggest you update the instruction to:

"Copy the token that appears on the screen and save it. You will not be able to retrieve the token once you close this prompt."

Author: Updated

once you close this prompt, Docker will never show the token again.

      ![Copy Token](images/hub-copy-token.png){:width="700px"}

## Use an access token

You can use an access token anywhere that requires your Docker Hub
password.

When logging in from your Docker CLI client (`docker login --username <username>`),
omit the password in the login command. Instead, enter your token when asked for a password.

   > **Note**
   >

>    >If you have 2FA enabled, you must use a personal access token when logging in from the Docker CLI. If you don't have it enabled, this is an optional (but more secure) method of authentication.

Reviewer: If you have 2FA enabled, you must use a personal access token when logging in from the Docker CLI. 2FA is an optional, but more secure method of authentication.

Author: Updated

Docker Hub lets you create personal access tokens as alternatives to your password. You can use tokens to access Hub images from the Docker CLI.

Using personal access tokens provides some advantages over a password:
* You can investigate the last usage of the access token and disable or delete it if you find any suspicious activity.
* When logged in with an access token, you can't perform any admin activity on the account, including changing the password.

>**Important**
>
> Treat access tokens like your password and keep them secret. Store your
> tokens securely (for example, in a credential manager).
{: .important}

Access tokens are valuable for building integrations, as you can issue
multiple tokens &ndash; one for each integration &ndash; and revoke them at
any time.

   >**Note**
   >

>    > If you have [two-factor authentication (2FA)](/docker-hub/2fa) enabled on

Reviewer 2: ```suggestion
   > If you have [two-factor authentication (2FA)](2fa/index.md) enabled on
```

Reviewer 2: Actually; looks like this is no longer in a numbered list as well so also doesn't need indenting;

```markdown
> **Note**
>
> If you have [two-factor authentication (2FA)](2fa/index.md) enabled on
> your account, you must create at least one personal access token. Otherwise,
> you will be unable to log in to your account from the Docker CLI.
```

    | Features   | Free | Pro | Team |
    | --------------------- | ---- | ----- | ----- |
    | Personal Access Tokens (read, write, delete scope only)*    |  x |   x  |   x  |
    | Scoped Personal Access Tokens  |      | x   |   x   |
    | Unlimited Personal Access Tokens   |      |       |   x   |

      >*You can create 1 access token as part of the Free plan, 5 tokens per the Pro plan, and Unlimited tokens for the Team plan.
      For more information about a Pro or a Team plan, see [Docker Pricing](https://www.docker.com/pricing?utm_source=docker&utm_medium=webreferral&utm_campaign=docs_driven_upgrade){: target="_blank" rel="noopener" class="_"}.

6. Copy the token that appears on the screen and save it. You will not be able to retrieve the token once you close this prompt.

      ![Copy Token](images/hub-copy-token.png){:width="700px"}

## Use an access token

You can use an access token anywhere that requires your Docker Hub
password.

When logging in from your Docker CLI client (`docker login --username <username>`), omit the password in the login command. Instead, enter your token when asked for a password.

>    > **Note**
>    >
>    >If you have 2FA enabled, you must use a personal access token when logging in from the Docker CLI. 2FA is an optional, but more secure method of authentication.

Reviewer 2: It probably renders ok, but this is no longer part of a (bullet/numbered) list, so should not be indented. Perhaps also (although we mention it earlier) include a link to the 2FA page?

```suggestion
> **Note**
>
> If you have [two-factor authentication (2FA)](2fa/index.md) enabled, you must
> use a personal access token when logging in from the Docker CLI. 2FA is an
> optional, but more secure method of authentication.
```

## Modify existing tokens

You can rename, activate, deactivate, or delete a token as needed.

> 1. Access your tokens under **[Account Settings](https://hub-stage.docker.com/settings/general) > [Security](https://hub-stage.docker.com/settings/security)**.

Reviewer 2: Looks like these still point to the staging domain;

```suggestion
1. Access your tokens under **[Account Settings](https://hub.docker.com/settings/general) > [Security](https://hub.docker.com/settings/security)**.
```

Do we want these to open in a new window?

```suggestion
1. Access your tokens under **[Account Settings](https://hub.docker.com/settings/general){: target="_blank" rel="noopener" class="_"} > [Security](https://hub.docker.com/settings/security){: target="_blank" rel="noopener" class="_"}**.
```

Reviewer 2: Wah, sorry for the repeated comments. Looking at this on my machine, and it looks like we want to point users to the "security" tab here, so we may not need the first link ("general"), so this could be a single link with `Account Settings > Security` as caption;

```suggestion
1. Access your tokens under **[Account Settings > Security](https://hub.docker.com/settings/security){: target="_blank" rel="noopener" class="_"}**.
```

2. Select a token and click **Delete** or **Edit**, or use the menu on
the far right of a token row to bring up the edit screen. You can also
select multiple tokens to delete at once.

>       ![Delete or Edit](images/hub-delete-edit-token.png){: {:width="700px"}

Reviewer 2: Looks like a duplicate `{:` here

```suggestion
      ![Delete or Edit](images/hub-delete-edit-token.png){:width="700px"}
```

      >*You can create 1 access token as part of the Free plan, 5 tokens per the Pro plan, and Unlimited tokens for the Team plan.
      For more information about a Pro or a Team plan, see [Docker Pricing](https://www.docker.com/pricing?utm_source=docker&utm_medium=webreferral&utm_campaign=docs_driven_upgrade){: target="_blank" rel="noopener" class="_"}.

6. Copy the token that appears on the screen and save it. You will not be able to retrieve the token once you close this prompt.

      ![Copy Token](images/hub-copy-token.png){:width="700px"}

## Use an access token

You can use an access token anywhere that requires your Docker Hub
password.

When logging in from your Docker CLI client (`docker login --username <username>`), omit the password in the login command. Instead, enter your token when asked for a password.

> **Note**
>
> If you have [two-factor authentication (2FA)](2fa/index.md) enabled, you must
> use a personal access token when logging in from the Docker CLI. 2FA is an
> optional, but more secure method of authentication.

## Modify existing tokens

You can rename, activate, deactivate, or delete a token as needed.

> 1. Access your tokens under **[Account Settings](https://hub.docker.com/settings/general) > [Security](https://hub.docker.com/settings/security)**.

Reviewer 2: This probably went lost in my other comments 😅 https://github.com/docker/docker.github.io/pull/13084#discussion_r673258138

Perhaps combine the two links here

```suggestion
1. Access your tokens under **[Account Settings > Security](https://hub.docker.com/settings/security){: target="_blank" rel="noopener" class="_"}**.
```

* You can investigate when an access token was used last, and disable or delete it if you find any suspicious activity.
* When logged in with an access token, you can't perform any admin activity on the account, including changing the password.

>**Warning**
>
> Treat access tokens like your password and keep them secret. Store your
> tokens securely (for example, in a credential manager).

> {: .warning }

Reviewer: Suggest we use "Important" attribute to this as it's not really a warning?

Reviewer: Could you update this to use the Important attribute, rather than a warning?

Author: I agree, updated in new commit

* You can investigate when an access token was used last, and disable or delete it if you find any suspicious activity.
* When logged in with an access token, you can't perform any admin activity on the account, including changing the password.

>**Warning**
>
> Treat access tokens like your password and keep them secret. Store your
> tokens securely (for example, in a credential manager).
{: .warning }

Access tokens are also useful in building integrations, since you can issue
multiple tokens &ndash; one for each integration &ndash; and revoke them at
any time.

   >**Note**
   >
   > If you have [two-factor authentication (2FA)](/docker-hub/2fa) enabled on
   > your account, you must create at least one personal access token. Otherwise,
   > you will be unable to log in to your account from the Docker CLI.

## Create an access token

You can create as many tokens as you need.

> 1. Log in to [hub.docker.com](https://hub.docker.com).

Reviewer: Could you add the target=_blank attribute here so the Hub page opens in a new tab?

Author: Updated

multiple tokens &ndash; one for each integration &ndash; and revoke them at
any time.

> **Note**: If you have [two-factor authentication (2FA)](/docker-hub/2fa) enabled on your account, you must create at least one personal access token. Otherwise, you will be unable to log in to your account from the Docker CLI.

Author: Updated, changes will reflect in next commit @thaJeztah

3. Select **Security > New Access Token**.

4. Add a description for your token. Use something that indicates where
the token is going to be used, or set a purpose for the token.

>       ![](images/PAT-Menu.png)

Reviewer: Could you add an alt text for the image in brackets?

>       ![](images/PAT-Menu.png)

Reviewer 2: Looks like this image is replacing `hub-create-token.png`. Could you rename the new file to be the same name as the old name, and replace the existing image?

Author: Updated! Thank you!

5. You can view the following permissions from the drop-down:

>  - Read, Write, Delete: lets you to manage team members. This is an admin privilege.

Author: @technicallyjosh The tokens are used to manage repositories only, not Docker Accounts? Should I make this a note instead?

>  - Read, Write, Delete: lets you to manage team members. This is an admin privilege.

Reviewer 3: The scopes are only for repos/images currently

>  - Read, Write, Delete: lets you to manage team members. This is an admin privilege.

Author: @technicallyjosh Thank you!!

>  - Read, Write, Delete: lets you to manage team members. This is an admin privilege.

Reviewer 3: Confirmed after discussions with product. Just noting here for record. New PAT scopes are not used outside of repositories/images.

>  - Read & Write: let you to push images to any repository managed by your account.

Reviewer: Should this be `lets you` for consistency?

 - Read Only: lets you to view, search, and pull images from any public repositories and any private repositories that you have access to.
 - Public Repo Read-only: lets you view, search, and pull images from any public repositories.

>  >Note:

Reviewer: Could you use the updated format
```
Could you use the following format for Notes?

> **Note**
>
> Text
```

3. Select **Security > New Access Token**.

4. Add a description for your token. Use something that indicates where
the token is going to be used, or set a purpose for the token.

>       ![PAT Menu](images/hub-create-token.png)

Author: Updated

5. You can view the following permissions from the drop-down:
 * Read, Write, Delete: lets you to manage team members. This is an admin privilege.
 * Read & Write: lets you to push images to any repository managed by your account.
 * Read Only: lets you to view, search, and pull images from any public repositories and any private repositories that you have access to.

>  * Public Repo Read-only: lets you view, search, and pull images from any public repositories.

Author: Updated

 **Note**:

>  All users have access to Read, Write, and Delete. Only paid users have access to Read & Write, Read-only, and Public Repo Read-only.

Author: Thank you! @thaJeztah  Updated with new format

6. Copy the token that appears on the screen. Make sure you do this now:
once you close this prompt, Docker will never show the token again.

>       ![Copy Token](images/hub-copy-token.png)

Reviewer 2: Same here (width):

```suggestion
      ![Copy Token](images/hub-copy-token.png){: style="max-width: 60%"}
```

Author: Updated

>       Treat access tokens like your password and keep them secret. Store your tokens securely (for example, in a credential manager).

Reviewer 2: Not new in this PR, but this warning doesn't stand out very well. I think this is important information, and perhaps should be mentioned at the top of the page. Perhaps we should move it to the introduction / start of the page and make it an admonition (warning)

3. Select **Security > New Access Token**.

4. Add a description for your token. Use something that indicates where
the token is going to be used, or set a purpose for the token.

      ![PAT Menu](images/hub-create-token.png){: style="max-width: 60%"}

5. You can view the following permissions from the drop-down:
     * Read, Write, Delete: lets you to manage team members. This is an admin privilege.
     * Read & Write: lets you to push images to any repository managed by your account.
     * Read Only: lets you to view, search, and pull images from any public repositories and any private repositories that you have access to.
     * Public Repo Read-only: lets you view, search, and pull images from any public repositories.

 >**Note**
 >
 > All users have access to Read, Write, and Delete. Only paid users have
 > access to Read & Write, Read-only, and Public Repo Read-only.

> 6. Copy the token that appears on the screen. Make sure you do this now:

Reviewer: This appears at list item 1 in the preview. Indenting the Note above should fix this issue

![image](https://user-images.githubusercontent.com/47779042/124263621-69cc3780-db2b-11eb-803c-0578cdceb262.png)

Reviewer: @jerae-duffin This issue still exists in the preview - https://deploy-preview-13084--docsdocker.netlify.app/docker-hub/access-tokens/#create-an-access-token

Author: Updated

the far right of a token row to bring up the edit screen. You can also
select multiple tokens to delete them all once.

>       ![](images/Generate-PAT.png)

Reviewer 2: perhaps use the existing naming conventions so that the files can be easily found together (e.g. `hub-generate-token.png` or  `hub-copy-token.png`)

In general; try to use lowercase filenames (for consistency, and because URLs are case-sensitive)

Reviewer 2: Also looks like `hub-edit-token.png` is no longer used now, so we should remove that image.

Author: @thaJeztah This image hub-edit-token.png is still applicable. Updated in next commit.

>       ![Modify Token](images/hub-edit-token.png)

Reviewer 2: Same here:

```suggestion
      ![Modify Token](images/hub-edit-token.png){: style="max-width: 60%"}
```

Author: Updated

If you have 2FA enabled, you must use a personal access token when logging in
from the Docker CLI. If you don't have it enabled, this is an optional (but
more secure) method of authentication.

## Modify existing tokens

You can rename, activate, deactivate, or delete a token as needed.

1. Access your tokens under **[Account Settings](https://hub-stage.docker.com/settings/general) > [Security](https://hub-stage.docker.com/settings/security)**.

2. Select a token and click **Delete** or **Edit**, or use the menu on
the far right of a token row to bring up the edit screen. You can also

> select multiple tokens to delete them all once.

Reviewer: >  to delete them all once.

Can we rephrase this

Author: Updated

>       ![Delete or Edit](images/hub-delete-edit-token.png){: {:width="700px"}}

Reviewer: Extra `}`

Author: Updated

   > **Note**
   >

>    > You can select multiple tokens and delete at onnce.

Reviewer: typo: once

We've also mentioned about deleting the tokens in 2. Wondering if we should we repeat this here?

Author: Removed this sentence
